### PR TITLE
.NET 6 update

### PR DESCRIPTION
**Pull request by mara-huminiuc**
-  Migrate Jobbr.Server.WebAPI from .netstand2.0 to .net6.0
-  Migrate Jobbr.Server.WebAPI from ASP.NET to ASP.NET Core
-  Migrate Jobbr.WebApi.Tests to .net6.0
-  Migrate Sandbox.JobRunner to .net6.0
-  Migrate Sandbox.JobServer to .net6.0
-  Remove Newtonsoft, Fody and LibLog

### CODE REVIEW

File: source/Jobbr.Client/JobbrClient.cs [hunk 1]
```diff
@@ -15,251 +15,251 @@ public class JobbrClient : IJobbrClient
 
         public JobbrClient(string backend)
         {
-            this.Backend = backend + (backend.EndsWith("/") ? string.Empty : "/");
-            this.httpClient = new HttpClient { BaseAddress = new Uri(this.Backend) };
+            Backend = backend + (backend.EndsWith("/") ? string.Empty : "/");
+            httpClient = new HttpClient { BaseAddress = new Uri(Backend) };
```
Comment (eXpl0it3r) [inline source/Jobbr.Client/JobbrClient.cs]: Fields should start with an underscore `_httpClient`

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 2]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
 
-            var hostingStarter = services.GetService<IHostingStarter>();
-            this.webHost = hostingStarter.Start(options);
+            builder.Services
+                .AddControllers()
+                .AddJsonOptions(options =>
+                {
+                    options.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+                    options.JsonSerializerOptions.PropertyNameCaseInsensitive = true;
+                    options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Can't you use the DefaultJsonOptions class here?

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 3]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
+                    options.JsonSerializerOptions.Converters.Add(new JsonTypeConverter<JobTriggerDtoBase>(_loggerFactory, "TriggerType", this.JobTriggerTypeResolver));
+                })
+                .AddApplicationPart(typeof(WebHost).Assembly);
 
-            Logger.InfoFormat($"Started OWIN-Host for WebAPI at '{this.configuration.BackendAddress}'.");
+            builder.Services.AddCors(options =>
+            {
+                options.AddPolicy("CorsPolicy", config => config
+                    .AllowAnyOrigin()
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Did we previously also allow any origin?

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 4]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
+            _webApp.MapControllers();
+
+            var backendAddressUri = new Uri(_configuration.BackendAddress);
+            _webApp.UsePathBase(backendAddressUri.AbsolutePath);
+            _webApp.Urls.Add(backendAddressUri.GetLeftPart(UriPartial.Authority));
+            _webApp.UseRouting();
+
+            _webApp.UseCors();
+
+            Task.FromResult(_webApp.StartAsync());
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: I guess we should have an async method here and call `await` instead

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 5]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
         }
 
         public void Stop()
         {
-            Logger.InfoFormat("Stopping OWIN-Host for Web-Endpoints'");
-
-            this.webHost?.Dispose();
-
-            this.webHost = null;
+            Task.FromResult(_webApp.StopAsync());
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Should use async & await

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 6]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
         }
 
         public void Dispose()
         {
-            this.webHost.Dispose();
-            this.webHost = null;
+            Task.FromResult(_webApp.StopAsync());
+            Task.FromResult(_webApp.DisposeAsync());
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Should implement `IAsyncDisposable`: https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 7]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
-            var services = (ServiceProvider)ServicesFactory.Create();
-            var options = new StartOptions()
+            if (_webApp != null)
+            {
+                throw new InvalidOperationException("The server has already been started.");
+            }
+
+            AssertBackendAddressIsValid();
+
+            var builder = WebApplication.CreateBuilder();
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Should this be a WebApplication and not just a WebHost?

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 8]
```diff
@@ -12,84 +18,128 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
-        /// </summary>
-        private readonly JobbrWebApiConfiguration configuration;
-
-        /// <summary>
-        /// The webhost that serves for the OWIN WebAPI component
-        /// </summary>
-        private IDisposable webHost;
+        private WebApplication _webApp;
+        private readonly ILogger _logger;
+        private readonly ILoggerFactory _loggerFactory;
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: I don't think the factory needs to be stored

File: source/Jobbr.WebApi.Tests/ExposeStorageProvider.cs [hunk 9]
```diff
@@ -1,17 +1,17 @@
-using Jobbr.ComponentModel.JobStorage;
+using Jobbr.ComponentModel.JobStorage;  
 using Jobbr.ComponentModel.Registration;
 
 namespace Jobbr.WebApi.Tests
 {
     public class ExposeStorageProvider : IJobbrComponent
     {
-        internal readonly IJobStorageProvider JobStorageProvider;
+        internal IJobStorageProvider JobStorageProvider { get; }
```
Comment (eXpl0it3r) [inline source/Jobbr.WebApi.Tests/ExposeStorageProvider.cs:8]: Why use a property?

File: source/Jobbr.WebApi.Tests/IntegrationTestBase.cs [hunk 10]
```diff
@@ -1,10 +1,11 @@
...
-using System.Net.Sockets;
-using Jobbr.ComponentModel.JobStorage;
+﻿using Jobbr.ComponentModel.JobStorage;
 using Jobbr.ComponentModel.Registration;
 using Jobbr.Server;
 using Jobbr.Server.Builder;
 using Jobbr.Server.WebAPI;
+using Microsoft.Extensions.Logging.Abstractions;
+using System.Net;
+using System.Net.Sockets;
```
Comment (eXpl0it3r) [inline source/Jobbr.WebApi.Tests/IntegrationTestBase.cs:8]: Don't think we need

```
using System.Net;
using System.Net.Sockets;
```

File: source/Sandbox.JobServer/Program.cs [hunk 11]
```diff
@@ -1,42 +1,50 @@
...
+            using ILoggerFactory loggerFactory = LoggerFactory.Create(builder => builder
+                .AddFilter(level => level >= LogLevel.Debug)
+                .AddConsole());
+
+            var jobbrBuilder = new JobbrBuilder(loggerFactory);
 
             jobbrBuilder.AddForkedExecution(config =>
             {
-                config.JobRunDirectory = "C:/temp";
+                config.JobRunDirectory = Path.GetTempPath();
```
Comment (eXpl0it3r) [inline source/Sandbox.JobServer/Program.cs:27]: 👌

File: source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs [hunk 12]
```diff
@@ -145,15 +144,25 @@ private static Type DefaultResolver(IEnumerable<Type> typesFromAllAssemblies, st
 
             if (!types.Any())
             {
-                throw new ArgumentException(string.Format("The Message-type '{0}' is not supported!", type));
+                throw new ArgumentException($"The Message-type '{type}' is not supported!");
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs]: Pass `nameof(type)` as second parameter

File: source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs [hunk 13]
```diff
@@ -145,15 +144,25 @@ private static Type DefaultResolver(IEnumerable<Type> typesFromAllAssemblies, st
...
             }
 
             if (types.Count() > 1)
             {
-                throw new ArgumentException(string.Format("multiple types for typename '{0}' found!", type));
+                throw new ArgumentException($"multiple types for typename '{type}' found!");
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs]: Pass `nameof(type)` as second parameter

File: source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs [hunk 14]
```diff
@@ -88,52 +87,52 @@ private static List<Type> GetTypesFromAllAssemblies()
...
 
             var typesFromAllAssemblies = GetTypesFromAllAssemblies();
 
-            var type = this.resolver(typesFromAllAssemblies, typeValue);
+            var type = _resolver(typesFromAllAssemblies, typeValue);
 
             if (type == null)
             {
-                throw new ArgumentException(string.Format("The cannot create object for empty type", typeValue));
+                throw new ArgumentException($"Cannot create object for empty type {typeValue}");
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/JsonTypeConverter.cs]: Pass `nameof(typeValue)` as second parameter

File: source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs [hunk 15]
```diff
@@ -5,11 +5,11 @@ namespace Jobbr.Server.WebAPI.Model
     /// <summary>
     /// The job trigger base.
     /// </summary>
-    public abstract class JobTriggerDtoBase
+    public class JobTriggerDtoBase
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs:8]: Why is this class no longer considered abstract?

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Lots of work, well done! 🥳 

A few minor comments here and there. What I'm not exactly sure about is your choice on WebApplication over WebHost. Can you elaborate on that a bit more?

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs (hunk 3 above)
Comment (mara-huminiuc) [reply source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: previously was EnableCorsAttribute("*", "*", "*"), so yes.

File: source/Jobbr.Server.WebAPI/Controller/DefaultController.cs [hunk 16]
```diff
@@ -23,21 +24,21 @@ public DefaultController(JobbrWebApiConfiguration configuration)
         /// </returns>
         [HttpGet]
         [Route("status")]
```
Comment (RKivioja) [inline source/Jobbr.Server.WebAPI/Controller/DefaultController.cs]: You can combine these attributes like this: [HttpGet("status")]

File: source/Jobbr.Server.WebAPI/Controller/JobController.cs [hunk 17]
```diff
@@ -1,114 +1,114 @@
...
 
             if (existingJob != null)
             {
-                return this.Conflict();
+                return Conflict();
             }
 
-            var job = new Job { UniqueName = dto.UniqueName, Title = dto.Title, Type = dto.Type, Parameters = dto.Parameters != null ? JsonConvert.SerializeObject(dto.Parameters) : null, };
-            this.jobManagementService.AddJob(job);
+            var job = new Job { UniqueName = dto.UniqueName, Title = dto.Title, Type = dto.Type, Parameters = dto.Parameters != null ? JsonSerializer.Serialize(dto.Parameters) : null, };
```
Comment (RKivioja) [inline source/Jobbr.Server.WebAPI/Controller/JobController.cs]: Missing the default serialization options?

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs (hunk 4 above)
Comment (mara-huminiuc) [reply source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: Start method is defined in IJobbrComponent, so it cannot be made async, unless we change the interface

File: source/Jobbr.Server.WebAPI/Infrastructure/JsonExtensions.cs [hunk 18]
```diff
@@ -8,7 +8,7 @@ public static object Deserialize(this string json)
         {
             if (string.IsNullOrWhiteSpace(json) == false)
             {
-                return JsonConvert.DeserializeObject(json);
+                return JsonSerializer.Deserialize<object>(json);
```
Comment (RKivioja) [inline source/Jobbr.Server.WebAPI/Infrastructure/JsonExtensions.cs]: Probably should also have the default options here.

File: source/Sanbox.JobRunner/Program.cs [hunk 19]
```diff
@@ -1,8 +1,9 @@
...
-using System.IO;
-using Jobbr.Runtime;
+﻿using Jobbr.Runtime;
 using Jobbr.Runtime.ForkedExecution;
+using Microsoft.Extensions.Logging;
 using Sandbox.JobRunner.Jobs;
+using System;
+using System.IO;
 
 namespace Sanbox.JobRunner
```
Comment (RKivioja) [inline source/Sanbox.JobRunner/Program.cs]: This ancient typo could probably be fixed.

File: source/Sandbox.JobServer/Program.cs [hunk 20]
```diff
@@ -1,42 +1,50 @@
...
 namespace Sandbox.JobServer
 {
-    class Program
+    public static class Program
     {
-        static void Main(string[] args)
+        public static void Main(string[] args)
         {
-            var jobbrBuilder = new JobbrBuilder();
+            using ILoggerFactory loggerFactory = LoggerFactory.Create(builder => builder
```
Comment (RKivioja) [inline source/Sandbox.JobServer/Program.cs]: using var loggerFactory

File: source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs (hunk 15 above)
Comment (mara-huminiuc) [reply source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs:8]: After removing Newtonsoft, there was the following error when deserializing this type: "Deserialization of reference types without parameterless constructor is not supported" and removing the abstract keyword was the fix.

File: source/Jobbr.WebApi.Tests/ExposeStorageProvider.cs (hunk 9 above)
Comment (mara-huminiuc) [reply source/Jobbr.WebApi.Tests/ExposeStorageProvider.cs:8]: it is accessed like ExposeStorageProvider.Instance.JobStorageProvider in the setup for the integration tests

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs (hunk 8 above)
Comment (mara-huminiuc) [reply source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs]: it is injected in JsonTypeConverter

File: source/Jobbr.WebApi.Tests/IntegrationTestBase.cs (hunk 10 above)
Comment (mara-huminiuc) [reply source/Jobbr.WebApi.Tests/IntegrationTestBase.cs:8]: NextFreeTcpPort method needs them

File: source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs (hunk 15 above)
Comment (eXpl0it3r) [reply source/Jobbr.Server.WebAPI.Model/JobTriggerDtoBase.cs:8]: Eh, I guess it's okay 😄

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs [hunk 21]
```diff
@@ -12,79 +13,81 @@ namespace Jobbr.Server.WebAPI.Infrastructure
...
         }
 
         public void Stop()
         {
-            Logger.InfoFormat("Stopping OWIN-Host for Web-Endpoints'");
-
-            this.webHost?.Dispose();
-
-            this.webHost = null;
+            Task.FromResult(_webHost.StopAsync());
```
Comment (eXpl0it3r) [inline source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs:66]: [Some say](https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c) that there are better methods than `Task.FromResult`, but I'm not sure. Always seems like a topic nobody has a good answer for

File: source/Sandbox.JobRunner/Jobs/ParameterizedJob.cs [hunk 22]
```diff
@@ -1,13 +1,13 @@
...
+﻿using System;
+using Sandbox.JobRunner.Jobs;
+
+namespace Sandbox.JobRunner.Jobs
+{
+    public class ParameterizedJob
+    {
+        public void Run(object jobParameters, RunParameter runParameters)
+        {
+            Console.WriteLine("Got the params {0} and {1}", jobParameters, runParameters);
```
Comment (eXpl0it3r) [inline source/Sandbox.JobRunner/Jobs/ParameterizedJob.cs]: Use string interpolation

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Might want to double check that this is the best way to call async functions from a non async context

File: source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs (hunk 21 above)
Comment (mara-huminiuc) [reply source/Jobbr.Server.WebAPI/Infrastructure/WebHost.cs:66]: You are right, there are multiple solutions out there, I still don't know if it's the best one but I think it should do.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**